packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -2,4 +2,617 @@
+to its set of allowed HTTP routes and command names. You do this by assigning the `newMethodMap`
+static variable in your driver class to an object of the same form as Appium's `routes.js` object.
+For example, here is the `newMethodMap` for the FakeDriver example driver:
+
+```js
+static newMethodMap = {
+  '/session/:sessionId/fakedriver': {
+    GET: {command: 'getFakeThing'},
+    POST: {command: 'setFakeThing', payloadParams: {required: ['thing']}},
+  },
+  '/session/:sessionId/fakedriverargs': {
+    GET: {command: 'getFakeDriverArgs'},
+  },
+};
+```
+
+In this example we're adding a few new routes and a total of 3 new commands. For more examples of
+how to define commands in this way, it's best to have a look through `routes.js`. Now all you need
+to do is implement the command handlers in the same way you'd implement any other Appium command.
+
+The downside of this way of adding new commands is that people using the standard Appium clients
+won't have nice client-side functions designed to target these endpoints. So you would need to
+create and release client-side plugins for each language you want to support (directions or
+examples can be found at the relevant client docs).
+
+An alternative to this way of doing things is to overload a command which all WebDriver clients
+have access to already: Execute Script. A common driver pattern is to create a magic prefix which
+the driver looks for at the beginning of any Execute Script invocation, and if it's present, to
+treat the rest of the script string as the name of a custom command to execute. This works in large
+part because for most platforms beyond web browsers, executing arbitrary JavaScript isn't a thing
+that makes sense.
+
+Let's say you are building a driver for stereo system called `soundz`, and you wanted to create
+a command for playing a song by name. You could expose this to your users in such a way that they
+call something like:
+
+```js
+// webdriverio example
+driver.executeScript('soundz: playSong', [{song: 'Stairway to Heaven', artist: 'Led Zeppelin'}]);
+```
+
+Then in your driver code you could simply look for `soundz:` at the beginning of your
+`execute` override, and do the appropriate thing:
+
+```js
+async execute(script, args) {
+  if (script === 'soundz: playSong') {
+    const {song, artist} = args[0];
+    // play the song based on song and artist details
+    return;
+  }
+  // otherwise do the normal thing, or throw a not yet implemented
+}
+```
+
+### Implement handling of Appium settings
+
+Appium users can send parameters to your driver via CLI args as well as via capabilities. But these
+cannot change during the course of a test, and sometimes users want to adjust parameters mid-test.
+Appium has a [Settings](../guides/settings.md) API for this purpose.
+
+To support settings in your own driver, first of all define `this.settings` to be an instance of
```

Comment:
Why don't we do this by default?

Reply:
ah, okay. 

in that case it seems like it'd make sense--in a utopian future where the API is perfect--to just have a function or property which defines the defaults, and the thing that instantiates `DeviceSettings` could just use it. (similarly to other things here)

Reply:
agreed.